riscos_toolbox/__init__.py

```diff
@@ -44,22 +44,54 @@ class Messages:
 _id_block       = IDBlock()
 _msgtrans_block = swi.block(4)
 
-def report_exception(e):
+
+def task_name():
+    try:
+        name_len = swi.swi('Toolbox_GetSysInfo', 'I0;..I', 0)
+        name_block = swi.block((name_len + 3) // 4)
```

Comment:
This should be + 4, I think. Otherwise, you're just rounding the length of the string to a word boundary, but you actually want to round the length of the string *and* terminator to the word boundary.

Reply:
<html><body>
<!--StartFragment-->

  | R2 holds size of buffer required (if R1 was 0)
-- | --


<!--EndFragment-->
</body>
</html>

however swi.block takes a number of words, so it needs rounding up to a number of words.